### PR TITLE
Show source code controls even if a file is not present

### PR DESCRIPTION
**Pull request by fscholdei**
## Proposed changes

If you switch to a branch in the source code view where the file does not exist, a 404 error is displayed. This error covers all control elements.

https://user-images.githubusercontent.com/45232454/120471772-6bc49e80-c3a5-11eb-96bf-a6c7caf5bb3d.mp4

This fix preserves the context and shows the error in the component where normally the content of the file is displayed, so that you can still change the branch and the path.

### Your checklist for this pull request

**Contributor**:
- [x] PR is well described and the description can be used as a commit message on squash
- [x] Related issues linked to PR if existing and labels set
- [x] Target branch is not master (in most cases develop should bet the target of choice) 
- [x] Code does not conflict with target branch
- [ ] New code is covered with unit tests
- [ ] New ui components are tested inside the storybook (module ui-components only) 
- [x] [Changelog entry file](https://github.com/scm-manager/changelog#changelog-entry-files) created in `gradle/changelog` or CHANGELOG.md is updated for plugins
- [ ] Documentation updated (only necessary for new features or changed behaviour)

**Reviewer**:
- [ ] The clean code principles are respected ([CleanCode](https://clean-code-developer.com/virtues/))
- [ ] All new code/logic is implemented on the right spot / "Should this be done here?"
- [ ] UI changes fits into the layout
- [ ] The UI views / components are responsive (mobile views)
- [ ] Correct translations are available

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

File: scm-ui/ui-webapp/src/repos/sources/containers/Content.tsx [hunk 1]
```diff
@@ -81,80 +76,82 @@ const BorderLessDiv = styled.div`
...
+    setCollapsed(!collapsed);
   };
 
-  handleExtensionError = (error: Error) => {
-    this.setState({
-      errorFromExtension: error
-    });
+  const handleExtensionError = (error: Error) => {
+    setErrorFromExtension(error);
   };
```
Comment (sdorra) [inline scm-ui/ui-webapp/src/repos/sources/containers/Content.tsx:105]: Why do we wrap a function with the exact same signature?

File: scm-ui/ui-webapp/src/repos/sources/containers/Content.tsx [hunk 2]
```diff
@@ -81,80 +76,82 @@ const BorderLessDiv = styled.div`
...
-    };
-  }
+  const wrapContent = (content: ReactNode) => {
+    return (
+      <>
+        <div className="panel">
+          {breadcrumb}
+          {content}
+        </div>
+        {errorFromExtension && <ErrorNotification error={errorFromExtension} />}
```
Comment (sdorra) [inline scm-ui/ui-webapp/src/repos/sources/containers/Content.tsx:94]: Is the same as just

```suggestion
        <ErrorNotification error={errorFromExtension} />
```

File: scm-ui/ui-webapp/src/repos/sources/containers/Content.tsx [hunk 3]
```diff
@@ -81,80 +76,82 @@ const BorderLessDiv = styled.div`
...
+export type SourceViewSelection = "source" | "annotations" | "history";
 
-class Content extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
+const Content: FC<Props> = ({ file, repository, revision, path, breadcrumb, error }) => {
+  const [t] = useTranslation("repos");
+  const [collapsed, setCollapsed] = useState(true);
+  const [selected, setSelected] = useState<SourceViewSelection>("source");
+  const [errorFromExtension, setErrorFromExtension] = useState<Error | null>(null);
```
Comment (sdorra) [inline scm-ui/ui-webapp/src/repos/sources/containers/Content.tsx:85]: Why not just?
```suggestion
  const [errorFromExtension, setErrorFromExtension] = useState<Error>();
```

Comment (fscholdei) [reply scm-ui/ui-webapp/src/repos/sources/containers/Content.tsx:85]: Thanks for all the good advice! 👍

**Pull request merged**